### PR TITLE
Add a new config to configure the host of the scheduler health check server

### PR DESCRIPTION
**Pull request by hussein-awala**
For security reasons, using "0.0.0.0" as a host is not a good practice, even for a health check server.

This PR creates a new config to configure the host of the scheduler health check server and uses the existing one as the default value.

related: #35615

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -2130,6 +2130,14 @@ scheduler:
...
       example: ~
       default: "False"
+    scheduler_health_check_server_host:
+      description: |
+        When you start a scheduler, airflow starts a tiny web server
+        subprocess to serve a health check on this host
+      version_added: 2.8.0
+      type: string
+      example: ~
+      default: "0.0.0.0"
```
Comment (potiuk) [inline airflow/config_templates/config.yml:2152]: Should the default **continue** to be 0.0.0.0 ? For me it looks like we might choose no default and use IP reverse-dnsed from hostname callable: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#hostname-callable maybe?). That would be quite a bit more secure default and only slightly risky when it comes to backwards compatibility.

**Comment by potiuk** [comment]
(and nice bandit to tell us so - BTW :)

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (hussein-awala) [reply airflow/config_templates/config.yml:2152]: I thought about it, but I think it's a breaking change.

Comment (potiuk) [reply airflow/config_templates/config.yml:2152]: A bit. But we do allow breaking changes "because security" ....

Comment (potiuk) [reply airflow/config_templates/config.yml:2152]: (just saying :D) . Not very strong on it

**Review by potiuk — APPROVED** [review]

**Review by amoghrajesh — COMMENTED** [review]
@hussein-awala generally the changes look fine, can you have some test coverage here: test_scheduler_command.py?

**Comment by hussein-awala** [comment]
Static checks failure is not related to this PR; I will try to fix it in a separate PR.

**Pull request merged**